watch.go

```diff
@@ -127,3 +127,25 @@ func (w WatchSet) watchMany(ctx context.Context) error {
 		return ctx.Err()
 	}
 }
+
+// WatchChan returns a channel that is used to wait for either the watch set
+// to trigger or for the context to be cancelled.
+func (w WatchSet) WatchChan(ctx context.Context) <-chan error {
+	// Create the outgoing channel
+	triggerCh := make(chan error)
```

Comment:
Buffer the channel to size 1 so that when the goroutine sends data, it never blocks and results in a hung goroutine. This is assuming you accept my feedback on removing the for loop.